modules/drivers/druid/test/metabase/driver/druid/client_test.clj

```diff
@@ -57,3 +57,35 @@
               (or (when (instance? java.net.ConnectException e)
                     (throw e))
                   (some-> (.getCause e) recur)))))))))
+
+(defn- test-request
+  ([request-fn]
+   (test-request request-fn true))
+  ([request-fn basic-auth?]
+   (try
+     (request-fn "http://localhost:8082/druid/v2"
+       :auth-enabled basic-auth?
+       :auth-username "nbotelho"
+       :auth-token "12345678910")
+     (catch Exception e
+       (ex-data e)))))
+
+(defn- get-auth-header [m]
+  (select-keys (:request-options m) [:basic-auth]))
+
+(deftest basic-auth-test
+  (let [get-request    (test-request druid.client/GET)
+        post-request   (test-request druid.client/POST)
+        delete-request (test-request druid.client/DELETE)]
+    (is (= {:basic-auth "nbotelho:12345678910"}
+          (get-auth-header get-request)
+          (get-auth-header post-request)
+          (get-auth-header delete-request)) "basic auth header included with successfully"))
+
+  (let [no-auth-basic false
+        get-request    (test-request druid.client/GET no-auth-basic)
+        post-request   (test-request druid.client/POST no-auth-basic)
+        delete-request (test-request druid.client/DELETE no-auth-basic)]
+    (is (= (contains? (get-auth-header get-request)    :basic-auth)
+           (contains? (get-auth-header post-request)   :basic-auth)
+           (contains? (get-auth-header delete-request) :basic-auth)) "basic auth header not included")))
```

Comment:
This isn't really testing whether or not the basic auth info is included in this case. They could all be returning `true` and this test would still pass.